src/main/java/edu/kit/datamanager/repo/util/validators/impl/HandleNetValidator.java

```
@@ -0,0 +1,156 @@
+     *
+     * @param prefix the handle prefix
+     * @param suffix the handle suffix
+     * @return true if the record is downloadable
+     */
+    private boolean isDownloadable(String prefix, String suffix) {
+        return isDownloadable("http://hdl.handle.net/api/handles", prefix, suffix);
+    }
+
+    /**
+     * This method uses regex to get the prefix, suffix and the server address out of a http or https URL.
+     * After this it uses the isDownloadable method to check whether the input is downloadable.
+     *
+     * @param url to validate
+     * @return true if the input is valid and downloadable.
+     */
+    private boolean isValidHTTPURL(String url) {
+        String regex = "(http|https)://(.+)/([A-Za-z0-9.]+)/([A-Za-z0-9.]+)";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(url);
+        if (matcher.find()) {
+            return isDownloadable((matcher.group(1) + "://" + matcher.group(2)), matcher.group(3), matcher.group(4));
+        } else throw new BadArgumentException("Invalid input");
+    }
+
+    /**
+     * This method uses regex to check whether unallowed characters are used and to extract the prefix and suffix.
+     * After this it uses the isDownloadable method to check whether the input is downloadable.
+     *
+     * @param handle to validate
+     * @return true if the input is valid and downloadable.
+     */
+    private boolean isValidHandle(String handle) {
+        String regex = "([A-Za-z0-9.]+)/([A-Za-z0-9.]+)";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(handle);
+        if (matcher.find()) {
+            return isDownloadable(matcher.group(1), matcher.group(2));
+        } else throw new BadArgumentException("Invalid input");
+    }
+
+    private static void ignoringExc(Runnable r) {
+        try {
+            r.run();
+        }
+        catch (Exception ignored) {
+        }
```

I don't like empty catch blocks. At least a message at debug level should be logged.

---

In this case the exception is unimportant and shouldn't be processed by anything.